relay-server/src/services/processor.rs

```diff
@@ -2627,7 +2627,14 @@ impl Service for EnvelopeProcessorService {
     type Interface = EnvelopeProcessor;
 
     fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
-        let thread_count = self.inner.config.cpu_concurrency();
+        // Adjust thread count for small cpu counts to not have too many idle cores
+        // and distribute workload better.
+        let thread_count = match self.inner.config.cpu_concurrency() {
+            0 | 1 => 1,
+            2 | 3 => 2,
+            4 => 3,
+            conc => conc - 2,
```

Comment:
Took me a while to understand the assignment, would this be easier to undestand?

```suggestion
            0 | 1 | 2 @ conc => conc.max(1),
            3 | 4 @ conc => conc - 1,
            conc => conc - 2,
```

(could also be assigned to a variable before match)